packages/checkout/widgets-lib/src/widgets/swap/functions/swapFees.ts

```diff
@@ -0,0 +1,67 @@
+import { Amount, Token, TransactionResponse } from '@imtbl/dex-sdk';
+import { TFunction } from 'i18next';
+import { BigNumber, utils } from 'ethers';
+import { CryptoFiatState } from '../../../context/crypto-fiat-context/CryptoFiatContext';
+import { calculateCryptoToFiat, tokenValueFormat } from '../../../lib/utils';
+
+type FormattedFee = {
+  label: string;
+  fiatAmount: string;
+  amount: string;
+  prefix?: string;
+  token: Token;
+};
+
+/**
+ * Formats a quote into a list of fees for the fee drawer
+ * @param swapQuote
+ * @param cryptoFiatState
+ * @param t
+ */
+export const formatSwapFees = (
+  swapQuote: TransactionResponse,
+  cryptoFiatState: CryptoFiatState,
+  t: TFunction,
+): FormattedFee[] => {
+  const fees: FormattedFee[] = [];
+  if (!swapQuote.swap) return fees;
+
+  const addFee = (estimate: Amount | undefined, label: string, prefix: string = '~ ') => {
+    const value = BigNumber.from(estimate?.value ?? 0);
+    if (estimate && value.gt(0)) {
+      const formattedFee = utils.formatUnits(value, estimate.token.decimals);
+      fees.push({
+        label,
+        fiatAmount: `~ ${t('drawers.feesBreakdown.fees.fiatPricePrefix')}${calculateCryptoToFiat(
```

Comment:
NOT A BLOCKER
@dreamoftrees this could use an interpolation

Reply:
Yep good call @jhesgodi I might leave it for a future iteration